export/export.go

```diff
@@ -0,0 +1,275 @@
+	Pods            []corev1.Pod                              `json:"podList"`
+	Nodes           []corev1.Node                             `json:"nodeList"`
+	Pvs             []corev1.PersistentVolume                 `json:"pvList"`
+	Pvcs            []corev1.PersistentVolumeClaim            `json:"pvcList"`
+	StorageClasses  []storagev1.StorageClass                  `json:"storageClassList"`
+	SchedulerConfig *v1beta2config.KubeSchedulerConfiguration `json:"schedulerConfig"`
+}
+
+type ResourcesApplyConfiguration struct {
+	Pods            []v1.PodApplyConfiguration                     `json:"podList"`
+	Nodes           []v1.NodeApplyConfiguration                    `json:"nodeList"`
+	Pvs             []v1.PersistentVolumeApplyConfiguration        `json:"pvList"`
+	Pvcs            []v1.PersistentVolumeClaimApplyConfiguration   `json:"pvcList"`
+	StorageClasses  []confstoragev1.StorageClassApplyConfiguration `json:"storageClassList"`
+	SchedulerConfig *v1beta2config.KubeSchedulerConfiguration      `json:"schedulerConfig"`
+}
+
+type PodService interface {
+	List(ctx context.Context) (*corev1.PodList, error)
+	Apply(ctx context.Context, pod *v1.PodApplyConfiguration) (*corev1.Pod, error)
+}
+
+type NodeService interface {
+	List(ctx context.Context) (*corev1.NodeList, error)
+	Apply(ctx context.Context, nac *v1.NodeApplyConfiguration) (*corev1.Node, error)
+}
+
+type PersistentVolumeService interface {
+	List(ctx context.Context) (*corev1.PersistentVolumeList, error)
+	Apply(ctx context.Context, persistentVolume *v1.PersistentVolumeApplyConfiguration) (*corev1.PersistentVolume, error)
+}
+
+type PersistentVolumeClaimService interface {
+	Get(ctx context.Context, name string) (*corev1.PersistentVolumeClaim, error)
+	List(ctx context.Context) (*corev1.PersistentVolumeClaimList, error)
+	Apply(ctx context.Context, persistentVolumeClaime *v1.PersistentVolumeClaimApplyConfiguration) (*corev1.PersistentVolumeClaim, error)
+}
+
+type StorageClassService interface {
+	List(ctx context.Context) (*storagev1.StorageClassList, error)
+	Apply(ctx context.Context, storageClass *confstoragev1.StorageClassApplyConfiguration) (*storagev1.StorageClass, error)
+}
+
+type SchedulerService interface {
+	GetSchedulerConfig() *v1beta2config.KubeSchedulerConfiguration
+	RestartScheduler(cfg *v1beta2config.KubeSchedulerConfiguration) error
+}
+
+func NewResourcesService(client clientset.Interface, pods PodService, nodes NodeService, pvs PersistentVolumeService, pvcs PersistentVolumeClaimService, storageClasss StorageClassService, schedulers SchedulerService) *Service {
+	return &Service{
+		client:              client,
+		podService:          pods,
+		nodeService:         nodes,
+		pvService:           pvs,
+		pvcService:          pvcs,
+		storageClassService: storageClasss,
+		schedulerService:    schedulers,
+	}
+}
+
+// Get all resources from each service.
+func (s *Service) get(ctx context.Context) (*Resources, error) {
+	getCtx := context.Background()
```

Comment:
Why do you create new context? (= why don't you use passed `ctx`?)

Reply:
I thought that we need to separate the Context. I just didn't understand it...
Fix it now.